tests/compile-fail/memleak.rs

```diff
@@ -0,0 +1,5 @@
+//error-pattern: the evaluated program leaked memory
+
+fn main() {
+    std::mem::forget(Box::new(42));
```

Comment:
Could you also add a test for an `Rc` cycle? There should be examples around back from before `mem::forget` was made safe, with the whole scoped threads discussion.